### PR TITLE
BE: Added the configuration provider and the database provider

### PR DESCRIPTION
**Pull request by Zlays**

### CODE REVIEW

**Review by gcoppola8 — APPROVED** [review]

File: backend/configs/config.go
Comment (guerinoni) [inline backend/configs/config.go:1]: Can you write tests for the pkg? At least we can avoid breaks

File: backend/main.go [hunk 1]
```diff
@@ -2,19 +2,33 @@ package main
...
 	"github.com/gin-contrib/cors"
 	"github.com/gin-gonic/gin"
 	socketio "github.com/googollee/go-socket.io"
+	"log"
 	"net/http"
 )
 
+func init() {
+	log.SetPrefix("[main]")
+}
```
Comment (guerinoni) [inline backend/main.go:16]: Not very happy of this... It make no sense to have logs based on pkg :)

File: backend/util/messages.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package util
+
+import "fmt"
+
+const STARTUP_ERROR = "Error during server startup: %v\n"
+
+func CustomError(err error) {
+	panic(fmt.Errorf(STARTUP_ERROR, err))
+}
```
Comment (guerinoni) [inline backend/util/messages.go:9]: Where those are used? And what is the purpose of this 🤔

**Review by guerinoni — CHANGES_REQUESTED** [review]
Can also with imperative + present simple for commits 
https://cbea.ms/git-commit/

File: backend/util/messages.go (hunk 2 above)
Comment (Zlays) [reply backend/util/messages.go:9]: It is currently used in the package 'configs'; if there should be an error during the application startup (loading configs and connecting to the db), the application must panic and not continue execution

Comment (Zlays) [reply backend/util/messages.go:9]: It might be better to rename the method or disassociate it from 'STARTUP_ERROR'

Comment (Zlays) [reply backend/util/messages.go:9]: removed

File: backend/main.go (hunk 1 above)
Comment (Zlays) [reply backend/main.go]: Removed

**Review by guerinoni — APPROVED** [review]
LGTML

**Pull request merged**